PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+import Foundation
+import StoreKit
+
+// MARK: Block definitions
+// NOTE: Any changes here must be reflected in PurchasesCoreSwift.h for ObjC compatibility.
+/**
+ Completion block for calls that send back a `PurchaserInfo`
+ */
+public typealias ReceivePurchaserInfoBlock = (PurchaserInfo?, Error?) -> Void
+
+/**
+ Completion block for `-[Purchases checkTrialOrIntroductoryPriceEligibility:completionBlock:]`
+ */
+public typealias ReceiveIntroEligibilityBlock = ([String: IntroEligibility]) -> Void
+
+/**
+ Completion block for `-[Purchases offeringsWithCompletionBlock:]`
+ */
+
+public typealias ReceiveOfferingsBlock = (Offerings?, Error?) -> Void
+
+/**
+ Completion block for `-[Purchases productsWithIdentifiers:completionBlock:]`
+ */
+public typealias ReceiveProductsBlock = ([SKProduct]) -> Void
+
+/**
+ Completion block for `-[Purchases purchaseProduct:withCompletionBlock:]`
+ */
+public typealias PurchaseCompletedBlock = (SKPaymentTransaction?, PurchaserInfo?, Error?, Bool) -> Void
+public typealias RCPurchaseCompletedBlock = PurchaseCompletedBlock
+
+/**
+ Deferred block for `purchases:shouldPurchasePromoProduct:defermentBlock:`
+ */
+public typealias DeferredPromotionalPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
+
+/**
+ * Deferred block for `-[Purchases paymentDiscountForProductDiscount:product:completion:]`
+ */
+@available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
+public typealias PaymentDiscountBlock = (SKPaymentDiscount?, Error?) -> Void
+
+/**
+ * `Purchases` is the entry point for Purchases.framework. It should be instantiated as soon as your app has a unique
+ * user id for your user. This can be when a user logs in if you have accounts or on launch if you can generate a random
+ * user identifier.
+ *  @warning Only one instance of Purchases should be instantiated at a time! Use a configure method to let the
+ *  framework handle the singleton instance for you.
+ */
+@objc(RCPurchases) public class Purchases: NSObject {
+
+    /**
+     * Delegate for `Purchases` instance. The delegate is responsible for handling promotional product purchases and
+     * changes to purchaser information.
+     */
+    @objc public var delegate: PurchasesDelegate? {
+        get { privateDelegate }
+        set {
+            privateDelegate = newValue
+            purchaserInfoManager.delegate = self
+            purchaserInfoManager.sendCachedPurchaserInfoIfAvailable(appUserID: appUserID)
+            Logger.debug(Strings.configure.delegate_set)
+        }
+    }
+
+    let subscriberAttributesManager: SubscriberAttributesManager
+
+    private weak var privateDelegate: PurchasesDelegate?
+    private let operationDispatcher: OperationDispatcher
+
+    /**
+     * Enable automatic collection of Apple Search Ads attribution. Disabled by default
+     */
+    @objc static var automaticAppleSearchAdsAttributionCollection: Bool = false
```

Comment:
all the `@objc` variables should be `public`, right?

Reply:
I mean, yeah, but how does the APITester project compile... and it does: `    automaticAppleSearchAdsAttributionCollection = [RCPurchases automaticAppleSearchAdsAttributionCollection];`

Reply:
(I'm updating all the vars for `public`)